# Reporte de descripción de los datos TB2 - Fundamentos de Data Science



---



## 1. Introducción

El presente apartado tiene como objetivo la inspección estructural de los datos recolectados. Se realizará una caracterización técnica de las variables, identificando tipos de datos y dimensiones para asegurar la consistencia del conjunto de datos antes de profundizar en el análisis exploratorio.

## 2. Inspección Estructural

In [1]:
import pandas as pd
import numpy as np
import json

df = pd.read_csv('MXvideos_cc50_202101.csv')

In [5]:
# Establecimiento de Volúmenes

print(f"Total de registros (filas): {df.shape[0]}")
print(f"Total de variables (columnas): {df.shape[1]}")

Total de registros (filas): 44043
Total de variables (columnas): 20


In [2]:
# Tabla 1: Vista previa de los primeros registros

print(df.head())

      video_id trending_date  \
0  SbOwzAl9ZfQ      17.14.11   
1  klOV6Xh-DnI      17.14.11   
2  6L2ZF7Qzsbk      17.14.11   
3  hcY52MFWMDM      17.14.11   
4  _OXDcGPVAa4      17.14.11   

                                               title  \
0                      Capítulo 12 | MasterChef 2017   
1  ALEXA EX-INTEGRANTE DEL GRUPO TIMBIRICHE RENUN...   
2             LOUIS CKAGÓ - EL PULSO DE LA REPÚBLICA   
3   Sismo de 6.7 sacude Costa Rica 12 Noviembre 2017   
4           DOG HACKS | MUSAS LESSLIE LOS POLINESIOS   

              channel_title  category_id      publish_time  \
0           MasterChef 2017         24.0   13/11/2017 1:06   
1  Micky Contreras Martinez         22.0   13/11/2017 0:11   
2  El Pulso De La República         25.0  13/11/2017 12:00   
3                  Casanare         25.0  12/11/2017 22:47   
4                     Musas         26.0  13/11/2017 14:17   

                                                tags     views    likes  \
0  MasterChef Junior 2

Se observa una estructura tabular compuesta por variables categóricas `(título, canal)` y numéricas `(vistas, likes, dislikes)`, lo cual concuerda con la naturaleza un dataset de tendencias.

In [10]:
# Análisis de tipos de datos y completitud inicial.

print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 44043 entries, 0 to 44042
Data columns (total 20 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   video_id                43959 non-null  object 
 1   trending_date           40593 non-null  object 
 2   title                   40510 non-null  object 
 3   channel_title           40501 non-null  object 
 4   category_id             40451 non-null  float64
 5   publish_time            40451 non-null  object 
 6   tags                    40480 non-null  object 
 7   views                   40451 non-null  float64
 8   likes                   40451 non-null  float64
 9   dislikes                40451 non-null  float64
 10  comment_count           40451 non-null  float64
 11  thumbnail_link          40471 non-null  object 
 12  comments_disabled       40451 non-null  object 
 13  ratings_disabled        40451 non-null  object 
 14  video_error_or_removed  40451 non-null

None

In [9]:
# Resumen de medidas de tendencia central y dispersión.

display(df.describe())

,category_id,views,likes,dislikes,comment_count,lat,lon
count,40451.000000,4.045100e+04,4.045100e+04,4.045100e+04,40451.000000,44043.000000,44043.000000
mean,21.003140,3.423820e+05,1.586184e+04,7.471604e+02,2039.660008,21.481515,-100.242120
std,5.878995,1.714691e+06,8.108987e+04,1.095358e+04,13938.031797,3.502304,6.018379
min,1.000000,1.570000e+02,0.000000e+00,0.000000e+00,0.000000,16.429991,-116.009260
25%,20.000000,1.681300e+04,2.990000e+02,1.700000e+01,42.000000,18.920381,-103.500024
50%,24.000000,5.697300e+04,1.246000e+03,6.300000e+01,196.000000,20.379982,-99.669999
75%,24.000000,2.068940e+05,7.226000e+03,2.670000e+02,885.000000,22.733353,-98.230010
max,43.000000,1.009124e+08,4.470923e+06,1.353667e+06,905925.000000,30.764051,-86.711455


In [8]:
# Resumen técnico de variables categóricas

display(df.describe(include=['object']))

,video_id,trending_date,title,channel_title,publish_time,tags,thumbnail_link,comments_disabled,ratings_disabled,video_error_or_removed,description,state,geometry
count,43959,40593,40510,40501,40451,40480,40471,40451,40451,40451,36246,44043,44043
unique,34250,272,33821,7005,29061,23647,33949,2,2,2,26727,32,32
top,\n,18.26.05,"Bachatas 2018 Romanticas - Prince Royce, Shaki...",Cracks,22/03/2018 23:00,[none],Don Pollo,FALSO,FALSO,FALSO,SUSCRIBETE es FACIL http://goo.gl/aLNQ6X\nIMP...,Morelos,POINT (-99.23999963999999 18.92110476)
freq,794,200,30,186,17,7685,17,40011,39817,40427,93,1441,1441


## 3. Conclusiones

Tras la inspección técnica del dataset "Trending YouTube Video Statistics", se han derivado las siguientes conclusiones fundamentales:

*  Volumen y estructura: El conjunto de datos comprende un total de 44,043 registros y 20 variables, proporcionando la escala operativa y representatividad necesarias para realizar un análisis robusto de las tendencias en la plataforma.

* Significado y formato: La estructura técnica se compone principalmente de variables de tipo object (categóricas y textuales) y float64 (métricas numéricas). La identificación de una llave primaria `(video_id)` junto con métricas clave de engagement (como` views, likes y dislikes` entre otras) establece la base semántica requerida para las etapas de procesamiento y modelado predictivo.

A continuación, se detalla el diccionario de datos, el cual especifica la naturaleza y el significado de cada campo analizado:

| Variable | Tipo Inicial | Significado |
|----------|--------------|-------------|
| `video_id` | `object` | Identificador único alfanumérico del video. |
| `trending_date` | `object` | Fecha en la que el video alcanzó tendencia. |
| `title` | `object` | Título del video. |
| `channel_title` | `object` | Nombre del canal que publicó el video. |
| `category_id` | `float64` | Código numérico asociado a la categoría del video. |
| `publish_time` | `object` | Fecha y hora exacta de publicación del video. |
| `tags` | `object` | Etiquetas o palabras clave asociadas al video. |
| `views` | `float64` | Conteo total de visualizaciones acumuladas. |
| `likes` | `float64` | Cantidad total de "me gusta" recibidos. |
| `dislikes` | `float64` | Cantidad total de "no me gusta" recibidos. |
| `comment_count` | `float64` | Conteo total de comentarios en el video. |
| `thumbnail_link` | `object` | URL de la imagen en miniatura del video. |
| `comments_disabled` | `object` | Indicador booleano de si los comentarios están desactivados. |
| `ratings_disabled` | `object` | Indicador booleano de si las valoraciones están desactivadas. |
| `video_error_or_removed` | `object` | Indicador booleano de si el video fue eliminado o presenta error. |
| `description` | `object` | Descripción detallada proporcionada por el autor del video. |
| `state` | `object` | Estado geográfico asociado a la tendencia del video. |
| `lat` | `float64` | Latitud de la ubicación geográfica del registro. |
| `lon` | `float64` | Longitud de la ubicación geográfica del registro. |
| `geometry` | `object` | Representación geométrica (tipo `POINT`) de la ubicación. |

